toponetx/classes/cell_complex.py

```diff
@@ -870,6 +870,114 @@ def get_filtration(self, name: str) -> dict[Union[Hashable, tuple], Any]:
                 d.update(i)
         return d
 
+    def set_node_attributes(
+        self,
+        values: Union[dict[Hashable, dict], dict[Hashable, Any]],
+        name: Optional[str],
+    ):
+        """Set node attributes.
+
+        Parameters
+        ----------
+        values :  dict
+            either contains node -> value (if `name` is specified)
+            or nested dict with node -> (attribute -> value) (if `name == None`)
+        name : str, optional
+
+        Returns
+        -------
+        None.
+
+        Examples
+        --------
+        >>> G = nx.path_graph(3)
+        >>> CX = CellComplex(G)
+        >>> CX.add_cell([1,2,3,4], rank=2)
+        >>> d={ 1: { 'color':'red','attr2':1 },2: {'color':'blue','attr2':3 } }
+        >>> CX.set_node_attributes(d)
+        >>> CX.nodes[1]['color']
+        'red'
+        """
+        self.set_cell_attributes(values, rank=0, name=name)
+
+    def get_node_attributes(self, name: str) -> dict[tuple, Any]:
+        """Get node attributes.
+
+        Parameters
+        ----------
+        name : str
+
+        Returns
+        -------
+        attr :  dict of node -> value of attribute `name`
+            nodes without the given attribute are not in the dictionary.
```

Comment:
The documentation of the returned value is not as mandated by NumPy's documentation style.

1. The returned value doesn't have a name, so remove the `attr : ` from the first line.
2. The first line only describes the type, anything else is part of the comment starting in the second line. That the mapping is from nodes to value is not part of the type.

Reply:
Ah, it is like this everywhere. I also fixed other occasions I came accross